project/src/ui/hud/SettingPane.gd

```diff
@@ -30,6 +30,8 @@ onready var save_btn: Button = $VBoxContainer/MarginContainer/VBoxContainer/HBox
 onready var world_list: OptionButton = $VBoxContainer/MarginContainer/VBoxContainer/HBoxContainer2/Worlds
 onready var sketches_label: Label = $VBoxContainer/MarginContainer/VBoxContainer/Sketches
 onready var boards_label: Label = $VBoxContainer/MarginContainer/VBoxContainer/Boards
+onready var compiler_lbl: Label = $VBoxContainer/MarginContainer/VBoxContainer/HBoxContainer3/Compiler
+onready var compiler_btn: MenuButton = $VBoxContainer/MarginContainer/VBoxContainer/HBoxContainer3/SelectCompiler
```

Comment:
Considered using OptionButton instead?

Reply:
Yes, this is what I had first. However, I think it is better this way because the list of compilers is only loaded from the backend when the user clicks on the button. When I had the option button I did not find a proper way of showing the user the saved compiler and at the same time not loading the backend data...

Reply:
I have now introduced an OptionButton. I realised that the compilers need to be loaded when the application starts to check whether the compiler from the profile still exists.